rust/windows-client/src-tauri/src/client/wintun_install.rs

```diff
@@ -26,20 +29,15 @@ pub(crate) enum Error {
     WriteFailed(io::Error),
 }
 
-/// Installs the DLL alongside the current exe, if needed
-/// The reason not to do it in the current working dir is that deep links may launch
-/// with a current working dir of `C:\Windows\System32`
-/// The reason not to do it in AppData is that learning our AppData path before Tauri
-/// setup is difficult.
-/// The reason not to do it in `C:\Program Files` is that running in portable mode
-/// is useful for development, even though it's not supported for production.
-pub(crate) fn ensure_dll() -> Result<(), Error> {
+/// Installs the DLL in %LOCALAPPDATA% and returns the DLL's absolute path
+///
+/// e.g. `C:\Users\User\AppData\Local\dev.firezone.client\data\wintun.dll`
+pub(crate) fn ensure_dll() -> Result<PathBuf, Error> {
     let dll_bytes = get_dll_bytes().ok_or(Error::PlatformNotSupported)?;
 
-    let path = tauri_utils::platform::current_exe()
-        .map_err(|_| Error::CurrentExePathUnknown)?
-        .with_file_name("wintun.dll");
-    tracing::debug!("wintun.dll path = {path:?}");
+    let path = dll_path()?;
+    std::fs::create_dir_all(path.with_file_name("")).map_err(|_| Error::CreateDirAll)?;
```

Comment:
```suggestion
    std::fs::create_dir_all(path.parent()).map_err(|_| Error::CreateDirAll)?;
```

Is the intention here for`with_file_name("")` to make sure we don't use the `parent` in case `path` is a directory?

[That doesn't seem to work](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=89ede11d6845d53446006b00b1e198b3)

Reply:
Just forgot `parent` existed